js/background/background.js

```diff
@@ -187,6 +187,63 @@ AugmentedSteamApi._progressingRequests = new Map();
 AugmentedSteamApi._earlyAccessAppIds_promise = null;
 
 
+class ContextMenu {
+    static init() {
+        ContextMenu.update();
+        chrome.contextMenus.onClicked.addListener(ContextMenu.onClick);
+    }
+
+    static onClick(info) {
+        switch (info.menuItemId) {
+            case "context_steam_store":
+                chrome.tabs.create({url: "https://store.steampowered.com/search/?term=" + encodeURIComponent(info.selectionText)});
+                break;
+            case "context_steam_market":
+                chrome.tabs.create({url: "https://steamcommunity.com/market/search?q=" + encodeURIComponent(info.selectionText)});
+                break;
+            case "context_itad":
+                chrome.tabs.create({url: "https://isthereanydeal.com/#/filter:&search/" + encodeURIComponent(info.selectionText) + ";/scroll:%23gamelist"});
+                break;
+            case "context_steamdb":
+                chrome.tabs.create({url: "https://steamdb.info/search/?q=" + encodeURIComponent(info.selectionText)});
+                break;
+            case "context_steamdb_instant":
+                chrome.tabs.create({url: "https://steamdb.info/instantsearch/?idx=steamdb&q=" + encodeURIComponent(info.selectionText)});
+                break;
+            case "context_steam_keys":
+                let steamkeys = info.selectionText.match(/[A-NP-RTV-Z02-9]{5}(-[A-NP-RTV-Z02-9]{5}){2}/g);
+                if (!steamkeys || steamkeys.length === 0) {
```

Comment:
This is the most common form of steam key. It's not the only form:
https://web.archive.org/web/20160414152200/https://support.steampowered.com/kb_article.php?ref=7480-WUSF-3601

Reply:
Yes, but we wanna be strict and make sure we only detect correct steam keys. This way users could select a whole page and detect steam keys in that.